hil/cli/network.py

```diff
@@ -0,0 +1,81 @@
+"""Commands related to networks are in this module"""
+import click
+import sys
+from hil.cli.client_setup import setup_http_client
+
+C = setup_http_client()
+
+
+@click.group()
+def network():
+    """Commands related to network"""
+
+
+@network.command(name='create', short_help='Create a new network')
+@click.argument('network')
+@click.argument('owner')
+@click.option('--access')
+@click.option('--net_id')
+def network_create(network, owner, access, net_id):
+    """Create a link-layer <network>.  See docs/networks.md for details"""
+    if net_id is None:
+        net_id = ''
+    if access is None:
+        access = owner
+    C.network.create(network, owner, access, net_id)
+
+
+@network.command(name='delete')
+@click.argument('network')
+def network_delete(network):
+    """Delete a network"""
+    C.network.delete(network)
+
+
+@network.command(name='show')
+@click.argument('network')
+def network_show(network):
+    """Display information about network"""
+    q = C.network.show(network)
+    for item in q.items():
+        sys.stdout.write("%s\t  :  %s\n" % (item[0], item[1]))
+
+
+@network.command(name='list')
+def network_list():
+    """List all networks"""
+    q = C.network.list()
+    for item in q.items():
+        sys.stdout.write('%s \t : %s\n' % (item[0], item[1]))
+
+
+@network.group(name='project')
+def network_project():
+    """Commands that manipulate project and network relations"""
+
+
+@network_project.command('list')
+@click.argument('network')
+@click.option('--project')
+def list_network_attachments(network, project):
+    """Lists all the attachments from <project> for <network>
+
+    If <project> is `None`, lists all attachments for <network>
+    """
+    print C.network.list_network_attachments(network, project)
+
+
+@network_project.command(name='add')
+@click.argument('network')
+@click.argument('project')
+def network_grant_project_access(project, network):
```

Comment:
I really think this would scan better as e.g. `hil network grant-access --project manhattan --network foo`. There's not really a natural order to the arguments here, and having project be a sub-category of network for network & project calls feels clumsy to me.

Reply:
so then, `grant-access` becomes a command with project and network as optional arguments?

I was thinking of moving this call to `project` subcommands. Just like how we have, `hil project node add/remove` we can do `hil project network add/remove` if that makes sense.

I am fine doing either.

Reply:
Hm, I was hoping there was a way to have flags that were mandatory, but it doesn't seem like there is. (it obviously doesn't make sense for project/network to be optional for this call). We could keep them as positional, but still do `hil network grant-access ...`, which is nice in that it mirrors the structure of the client library.

Reply:
okay. I had done that in a previous commit and we decided to change because we had to make `grant-access` and `revoke-access` a subcommand under `network`. Just want to make sure that this is what you are suggesting before I do that again.
https://github.com/CCI-MOC/hil/pull/969/commits/5595dfabf6feb1512df722a09a40004689a2f153?diff=split

Reply:
If you're talking about this thread:

https://github.com/CCI-MOC/hil/pull/969#discussion_r173029097

The concern there was different, as you had a group for one command.